doc/nrf/releases_and_maturity/migration/nRF54H20_migration_2.7/migration_guide_2.6.99-cs2_to_2_7_environment.rst

```diff
@@ -0,0 +1,347 @@
+
+.. rst-class:: numbered-step
+
+Updating the FICR
+=================
+
+.. caution::
+   This step is required only if your nRF54H20 DK is version PCA10175 v0.7.x or v0.8.0 (ES3).
+   Jump to the next step if your DK is version ES4.
+
+After programming the SDFW and SCFW from the firmware bundle, you must update the Factory Information Configuration Registers (FICR) to correctly configure some trims of the nRF54H20 SoC.
+To update the FICR, you must run a J-Link script:
+
+1. Get the Jlink script that updates the FICR::
+
+      curl -LO https://files.nordicsemi.com/artifactory/swtools/external/scripts/nrf54h20es_trim_adjust.jlink
+
+#. Run the script:
+
+   * Linux and Mac OS::
+
+        JLinkExe -CommanderScript nrf54h20es_trim_adjust.jlink
+
+   * Windows::
+
+        jlink.exe -CommanderScript nrf54h20es_trim_adjust.jlink
+
+.. _migration_cs2_to_2_7_env_lcsrot:
+
+.. rst-class:: numbered-step
+
+Transitioning the nRF54H20 SoC to RoT
+=====================================
+
+Set the device to Root of Trust using the following command::
+
+      nrfutil device x-adac-lcs-change --life-cycle rot --serial-number <serial_number>
+
+#. Verify again the current lifecycle state of the nRF54H20::
+
+      nrfutil device x-adac-discovery --serial-number <serial_number>
+
+   The output will look similar to the following::
+
+      *serial_number*
+      adac_auth_version     1.0
+      vendor_id             Nordic VLSI ASA
+      soc_class             0x00005420
+      soc_id                [e6, 6f, 21, b6, dc, be, 11, ee, e5, 03, 6f, fe, 4d, 7b, 2e, 07]
+      hw_permissions_fixed  [00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00]
+      hw_permissions_mask   [01, 00, 00, 00, 87, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00]
+      psa_lifecycle         LIFECYCLE_ROT (0x2000)
+      sda_id                0x01
+      secrom_revision       0xad3b3cd0
+      sysrom_revision       0xebc8f190
+      token_formats         [TokenAdac]
+      cert_formats          [CertAdac]
+      cryptosystems         [Ed25519Sha512]
+      Additional TLVs:
+      TargetIdentity: [ff, ff, ff, ff, ff, ff, ff, ff]
+
+   The lifecycle state (``psa_lifecycle``) is now correctly set to *Root of Trust* (``LIFECYCLE_ROT (0x2000)``)
+
+#. After the LCS transition, reset the device::
+
+      nrfutil device reset --reset-kind RESET_PIN --serial-number <serial_number>
+
+nRF54H20 DK firmware bundle update
+**********************************
+
+.. caution::
+   Follow these steps only if your device is in lifecycle state ``RoT``.
+   For more information, see `Verify the LCS of the device`_.
+
+If your device is in ``RoT``, you can update to the newest release of the firmware bundle as follows:
+
+1. Download the `nRF54H20 firmware bundle v0.5.0`_.
+#. Move the :file:`ZIP` bundle to a folder of your choice and unzip the bundle.
+#. Purge all the domains::
+
+      nrfutil device x-boot-mode-set --boot-mode safe --traits jlink
```

Comment:
This should probably also make use of the `recover` command, similar to [nRF54H20_erase_UICR.txt](https://github.com/nrfconnect/sdk-nrf/pull/15760/files#diff-09f5954c9cab2509414b143fa91e074244162caefaf67af4efdf7bbe9db0d81b)

Reply:
Are you saying they they should also run the recover command after this one, or in place of this one? Should the recover command be run manually for each one of the cores? (app and radio) could you provide the exact syntax in this case? @ingeryp  FYI @miha-nordic

Reply:
Sorry about the slow response.
In the file I linked you this 4-line sequence is replaced with 2 calls to recover.
@Tiwalun could probably clarify this further if needed.

Reply:
The following command should be run:

```
nrfutil device recover --traits jlink --log-level trace --core Application
nrfutil device recover --traits jlink --log-level trace --core Network
```
This replaces the 4 commands which you currently have here.

Reply:
`--traits jlink` should probably be replaced with `--serial-number`, and the log level removed, to make it more consistent with the other commands:

```
nrfutil device recover --serial-number <serial_number> --core Application
nrfutil device recover --serial-number <serial_number> --core Network
```